### PR TITLE
Enable epochs handler integration part 6

### PR DESCRIPTION
**Pull request by sstanculeanu**
- refactored process/peer/process.go, heartbeat
- completely refactored:
   - BelowSignedThresholdEnableEpoch
   - StopDecreasingValidatorRatingWhenStuckEnableEpoch
   - HeartbeatDisableEpoch
   - TransactionSignedWithTxHashEnableEpoch
   
**Notes:**
1. In order to properly check if the old flags match the new methods, one who review should check old conditions from removed `EpochConfirmed` method, the value from config that was passed before, and the line from `EpochConfirmed` from `enableEpochsHandler.go` which uses the same config value.
2. No focus on long tests so far, they will be fixed on the last PR once all components use `enableEpochsHandler`.

### CODE REVIEW

File: epochStart/bootstrap/process.go [hunk 1]
```diff
@@ -990,7 +981,7 @@ func (e *epochStartBootstrap) updateDataForScheduled(
 		HeadersSyncer:        e.headersSyncer,
 		MiniBlocksSyncer:     e.miniBlocksSyncer,
 		TxSyncer:             e.txSyncerForScheduled,
-		ScheduledEnableEpoch: e.enableEpochs.ScheduledMiniBlocksEnableEpoch,
+		ScheduledEnableEpoch: e.coreComponentsHolder.EnableEpochsHandler().ScheduledMiniBlocksEnableEpoch(),
```
Comment (bogdan-rosianu) [inline epochStart/bootstrap/process.go:984]: why do you still need `ScheduledEnableEpoch` ?

File: integrationTests/factory/statusComponents/statusComponents_test.go [hunk 2]
```diff
@@ -62,7 +62,6 @@ func TestStatusComponents_Create_Close_ShouldWork(t *testing.T) {
 		managedCoreComponents.NodesShuffler(),
 		managedBootstrapComponents.ShardCoordinator().SelfId(),
 		managedBootstrapComponents.EpochBootstrapParams(),
-		managedBootstrapComponents.EpochBootstrapParams().Epoch(),
```
Comment (bogdan-rosianu) [inline integrationTests/factory/statusComponents/statusComponents_test.go:65]: handle `WaitingListFixEnableEpoch` as well

File: process/peer/process.go [hunk 3]
```diff
@@ -1021,7 +1000,7 @@ func (vs *validatorStatistics) updateValidatorInfoOnSuccessfulBlock(
 			peerAcc.SetConsecutiveProposerMisses(0)
 			newRating = vs.rater.ComputeIncreaseProposer(shardId, peerAcc.GetTempRating())
 			var leaderAccumulatedFees *big.Int
-			if vs.flagStakingV2Enabled.IsSet() {
+			if vs.enableEpochsHandler.IsStakingV2FlagEnabledForActivationEpochCompleted() {
```
Comment (bogdan-rosianu) [inline process/peer/process.go:1003]: looks good :+1:

File: epochStart/bootstrap/process.go (hunk 1 above)
Comment (sstanculeanu) [reply epochStart/bootstrap/process.go:984]: the epoch is still used [here](https://github.com/ElrondNetwork/elrond-go/blob/integrate_enable_epochs_handler_p6/epochStart/bootstrap/startInEpochScheduled.go#L223)..decided to only pass the epoch, not the entire component so the `startInEpochWithScheduledDataSyncer` won't be changed

File: integrationTests/factory/statusComponents/statusComponents_test.go (hunk 2 above)
Comment (sstanculeanu) [reply integrationTests/factory/statusComponents/statusComponents_test.go:65]: thanks, deleted the wrong line :man_facepalming:

**Review by bogdan-rosianu — APPROVED** [review]

File: process/peer/process_test.go [hunk 4]
```diff
@@ -1379,7 +1380,8 @@ func TestValidatorStatisticsProcessor_CheckForMissedBlocksMissedRoundsGreaterTha
 	arguments.PeerAdapter = peerAdapter
 	arguments.NodesCoordinator = nodesCoordinatorMock
 	arguments.MaxComputableRounds = 1
-	arguments.StopDecreasingValidatorRatingWhenStuckEnableEpoch = 2
+	stub, _ := arguments.EnableEpochsHandler.(*testscommon.EnableEpochsHandlerStub)
+	stub.IsStopDecreasingValidatorRatingWhenStuckFlagEnabledField = false
 	arguments.MaxConsecutiveRoundsOfRatingDecrease = 4
```
Comment (ssd04) [inline process/peer/process_test.go:1385]: maybe other name here?

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 5]
```diff
@@ -1191,8 +1186,3 @@ func createValidatorInfoFromBody(
 
 	return allValidatorInfo, nil
 }
-
-func (ihnc *indexHashedNodesCoordinator) updateEpochFlags(epoch uint32) {
-	ihnc.flagWaitingListFix.SetValue(epoch >= ihnc.waitingListFixEnableEpoch)
-	log.Debug("indexHashedNodesCoordinator: waiting list fix", "enabled", ihnc.flagWaitingListFix.IsSet())
-}
```
Comment (ssd04) [inline sharding/nodesCoordinator/indexHashedNodesCoordinator.go:1198]: if we remove this, it seems like this `flagWaitingListFix` var will not be set anymore, not sure if it safe to remove it

File: process/peer/process_test.go (hunk 4 above)
Comment (sstanculeanu) [reply process/peer/process_test.go]: renamed

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go (hunk 5 above)
Comment (sstanculeanu) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go:1198]: the flag was set but never used, that's why I completely removed it

Comment (ssd04) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go:1198]: nevermind, i've just seen that it was moved back in part 7

**Review by ssd04 — DISMISSED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**